sdk/core/azure-core/src/test/java/com/azure/android/core/util/ExpandableStringEnumTest.java

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.android.core.util;
+
+import org.junit.Test;
+
+import java.util.Collection;
+import java.util.Iterator;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertSame;
+import static org.junit.Assert.assertTrue;
+
+public class ExpandableStringEnumTest {
+    @Test
+    public void createInstance_fromString() {
+        assertNotNull(MyExpandableStringEnum.fromString("Test"));
+    }
+
+    @Test
+    public void createInstance_fromNullString() {
+        assertNull(MyExpandableStringEnum.fromString(null));
+    }
+
+    @Test
+    public void getExistingInstance_fromString() {
+        MyExpandableStringEnum myExpandableStringEnum = MyExpandableStringEnum.fromString("Test");
+
+        assertSame(myExpandableStringEnum, MyExpandableStringEnum.fromString("Test"));
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void throwException_whenCallingFromString_withNullClass() {
+        MyExpandableStringEnum.fromStringWithNullClass();
+    }
+
+    @Test
+    public void returnNull_whenTryingToGetInstanceOfPrivateClass_fromString() {
+        assertNull(MyPrivateExpandableStringEnum.fromString());
+    }
+
+    @Test
+    public void getExistingInstances() {
+        MyExpandableStringEnum myExpandableStringEnum = MyExpandableStringEnum.fromString("Test");
+        MyExpandableStringEnum otherMyExpandableStringEnum = MyExpandableStringEnum.fromString("Another");
+        Collection instances = MyExpandableStringEnum.values();
+
+        assertFalse(instances.isEmpty());
+        assertEquals(2, instances.size());
+
+        Iterator iterator = instances.iterator();
+
+        assertSame(myExpandableStringEnum, iterator.next());
+        assertSame(otherMyExpandableStringEnum, iterator.next());
+    }
+
+    @Test
+    public void convertToString() {
+        assertEquals("Test", MyExpandableStringEnum.fromString("Test").toString());
+    }
+
+    @Test
+    public void getHashCode() {
+        assertEquals(-2043209345, MyExpandableStringEnum.fromString("Test").hashCode());
+    }
```

Comment:
Same comment as the other `hashCode` test.

Reply:
See comment above for the same question.